packages/datagateway-download/src/downloadStatus/adminDownloadStatusTable.component.tsx

```diff
@@ -0,0 +1,600 @@
+                        onClick={() => {
+                          setIsRestoring(true);
+                          setTimeout(
+                            () =>
+                              adminDownloadDeleted(
+                                downloadItem.id as number,
+                                false,
+                                {
+                                  facilityName: settings.facilityName,
+                                  downloadApiUrl: settings.downloadApiUrl,
+                                }
+                              )
+                                .then(() => {
+                                  // Get the new status and isDeleted state of the download item
+                                  fetchAdminDownloads(
+                                    {
+                                      facilityName: settings.facilityName,
+                                      downloadApiUrl: settings.downloadApiUrl,
+                                    },
+                                    `WHERE UPPER(download.id) = ${downloadItem.id}`
+                                  ).then((downloads) => {
+                                    const formattedDownload = formatDownloads(
+                                      downloads
+                                    )[0];
+                                    downloadItem.status =
+                                      formattedDownload.status;
+                                    downloadItem.isDeleted =
+                                      formattedDownload.isDeleted;
+                                    setData(
+                                      data.map((download) =>
+                                        download.id === downloadItem.id
+                                          ? { ...download, ...downloadItem }
+                                          : download
+                                      )
+                                    );
+
+                                    if (
+                                      downloadItem.status !==
+                                      t('downloadStatus.restoring')
+                                    ) {
+                                      setIsRestoring(false);
+                                    }
+                                  });
+                                })
+                                .catch(() => {
+                                  setIsRestoring(false);
+                                }),
+                            100
+                          );
+                        }}
+                      >
+                        <Restore />
+                      </IconButton>
+                    );
+                  },
+
+                  function DeleteButton({ rowData }: TableActionProps) {
+                    const downloadItem = rowData as FormattedDownload;
+                    const [isDeleting, setIsDeleting] = React.useState(false);
+
+                    return (
```

Comment:
TopCAT hides this button if the download has already been deleted - is this possible? Otherwise we have a confusing button that does nothing

Reply:
I used the normal Download Status table as an example when I created the buttons hence why _Restore_ and _Delete_ are two separate buttons. It looks like it is possible to hide buttons if `null` is returned by the action function. This is how it behaves with some of the changes that I made just now:
![Admin Download table](https://user-images.githubusercontent.com/45173816/117461304-4dee5000-af45-11eb-862c-ad394c12a2f5.gif)

is this how we want it to work @louise-davies ?

Reply:
@VKTB that looks good and looks more like what TopCAT had and I think makes more sense.